apache/yunikorn-k8shim
# [YUNIKORN-2416] Cleanup replace directives

### What is this PR for?

- Cleanup unnecessarily golang.org/x/lint modules in replace directives. 
- Upgrade the the latest yunikorn-core/yunikorn-sheduler-interface
- Replace github.com/opencontainers/runc:v1.1.10 by v1.1.12 to fix [CVE-2024-21626](https://nvd.nist.gov/vuln/detail/CVE-2024-21626)

Note: There are 2 vulnerable dependencies existing before this PR, and it can't be fixed by replace directives. 
- go:gopkg.in/square/go-jose.v2:v2.6.0 is vulnerable (Cxb6dee8d5-b814)
 (From k8s.io/kubernetes@v1.29.2) (The packege is under public archive.)
- go:k8s.io/kubernetes:v1.29.2 is vulnerable (CVE-2020-8562)

And it required should be fixed in K8S.

### What type of PR is it?

### Todos
NA

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2416

### How should this be tested?
1. Run CVE check (or use your IDE's vulnerability analysis tool).
2. Run E2E test with this new dependencis

E2E passed in my side:
https://github.com/chenyulin0719/yunikorn-k8shim/actions/runs/8004883059

### Screenshots (if appropriate)
There are 2 existing CVE and can't be fixed by replace.
<img width="1432" alt="image" src="https://github.com/apache/yunikorn-k8shim/assets/26764036/662d2d69-bb0d-4129-9127-db91d7b855d7">

CVE check pass with govulncheck:
<img width="740" alt="image" src="https://github.com/apache/yunikorn-k8shim/assets/26764036/025acf79-10b7-466c-881f-c097da4fc182">



### Questions:
NA
